app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -86,6 +87,59 @@ export const snapKeyringBuilder = (
   removeAccountHelper: (address: string) => Promise<any>,
 ) => {
   const builder = (() => {
+    const snapAuthorshipHeader = (snapId: string): ResultComponent => {
+      return {
+        name: 'SnapAuthorshipHeader',
+        key: 'snapHeader',
+        properties: { snapId },
+      } as ResultComponent;
+    };
+
+    const getSnapName = (snapId: string): string => {
+      const subjectMetadata = controllerMessenger.call(
+        'SubjectMetadataController:getSubjectMetadata',
+        snapId,
+      );
+
+      return getSnapNameFromSubjectMetadata(snapId, subjectMetadata);
+    };
+
+    const showError = (
```

Comment:
This is a great refactor! Given that this file is getting massive I will suggest that we extract these util functions into the `snap-keyring/utils` folder. This will make rebases easier in the future.

Reply:
This is indeed a good point, lemme change that!

Reply:
I did split into `showResult.ts` file